pkg/client/events/replay_client.go

```diff
@@ -210,15 +224,32 @@ func (rClient *replayClient[T]) goPublishEvents(ctx context.Context, publishCh c
 		select {
 		case <-ctx.Done():
 			// If the context is done, exit the loop and stop processing events.
+			rClient.logger.Info().Msgf(
+				"🛑 Context canceled. Stopping event subscription for query: %s",
```

Comment:
This is what the user really cares about:

```suggestion
				"🛑 Stopping event subscription for query: %s",
```